src/composables/queries/usePoolSnapshotsQuery.ts

```diff
@@ -48,18 +56,57 @@ export default function usePoolSnapshotsQuery(
   const queryFn = async () => {
     if (!pool.value) throw new Error('No pool');
 
-    // TODO - remove this once coingecko supports wstEth
-    const tokens = pool.value.tokenAddresses.includes(addresses.wstETH)
-      ? [...pool.value.tokenAddresses, addresses.stETH]
-      : pool.value.tokenAddresses;
+    let tokens: string[] = [];
+
+    const isStablePhantomPool = isStablePhantom(pool.value.poolType);
+
+    if (isStablePhantomPool && pool.value.mainTokens != null) {
+      // TODO - remove this once coingecko supports linear tokens
+      tokens = pool.value.mainTokens;
+    } else if (pool.value.tokenAddresses.includes(addresses.wstETH)) {
+      // TODO - remove this once coingecko supports wstEth
+      tokens = [...pool.value.tokenAddresses, addresses.stETH];
+    } else {
+      tokens = pool.value.tokenAddresses;
+    }
+
+    const [historicalPrices, snapshots] = await Promise.all([
+      coingeckoService.prices.getTokensHistorical(tokens, days),
+      balancerSubgraphService.poolSnapshots.get(id, days)
+    ]);
+
+    if (isStablePhantomPool) {
+      const tokensWithoutPoolToken = tail(pool.value.tokens);
```

Comment:
@garethfuller do you think this is a good method? Essentially I'm assuming the first token is always the pool token, so the index number matters.